GVRf/Sample/simple-sample/src/org/gearvrf/simplesample/SampleViewManager.java

```diff
@@ -36,34 +57,194 @@ public void onInit(GVRContext gvrContext) {
-                mGVRContext, R.drawable.gearvr_logo));
+        // add the scene object to the scene graph
+        mMainScene.addSceneObject(mMainObject);
+        
+    }
+    
+    private void buildSceneObjects(){
 
-        // create a scene object (this constructor creates a rectangular scene
-        // object that uses the standard 'unlit' shader)
-        GVRSceneObject sceneObject = new GVRSceneObject(gvrContext, 4.0f, 2.0f,
-                texture);
+        mMainObject = buildImageObject();
+//        mMainObject = buildVideoObject();
+//        mMainObject = buildWebViewObject();
+//        mMainObject = buildCameraObject();
+//        mMainObject = buildModelObject();
 
         // set the scene object position
-        sceneObject.getTransform().setPosition(0.0f, 0.0f, -3.0f);
-
-        // add the scene object to the scene graph
-        scene.addSceneObject(sceneObject);
-
+        mMainObject.getTransform().setPosition(0.0f, 0.0f, -5.0f);
     }
 
+    
+    
     @Override
     public void onStep() {
+    	checkEyePointee();
     }
 
+    
+    
+    // 1. Create image object
+	private GVRSceneObject buildImageObject(){
+		
+		// Create a image scene object (this constructor creates a rectangular scene object)
+		GVRSceneObject imageObject = new GVRSceneObject(mGVRContext);
+		
+		
+		// Set Mesh from factory.
+        GVRRenderData renderData = new GVRRenderData(mGVRContext);
+        renderData.setMesh(mGVRContext.createQuad(4.0f, 2.0f));
+		
+		 // Load and set texture
+        GVRTexture texture = mGVRContext.loadTexture(new GVRAndroidResource(mGVRContext, R.drawable.gearvr_logo));
+
+        // Creat Material and set Texture.
+        GVRMaterial material = new GVRMaterial(mGVRContext, GVRShaderType.Unlit.ID);
+        material.setMainTexture(texture);
+        renderData.setMaterial(material);
+
+        imageObject.attachRenderData(renderData);
+        
+        
+
+        // Or, you can make this way. very easily.
+        // imageObject = new GVRSceneObject(mGVRContext, 4.0f, 2.0f, texture);
+        
+        return imageObject;
+	}
+	
+	
+	
+	// 2. Create video object
+	private GVRSceneObject buildVideoObject(){
+
+		// Create your own MediaPlayer.
+		// This MediaPlayer follow Android MediaPlayer.
+        MediaPlayer mediaPlayer = MediaPlayer.create(mGVRContext.getContext(), R.drawable.tron);
+        mediaPlayer.start();
+        
+        // From inside of GVRVideoSceneObject, refresh MediaPlayer automatically.
+		GVRVideoSceneObject videoObject = new GVRVideoSceneObject(mGVRContext, 4.0f,2.0f, mediaPlayer, GVRVideoType.MONO);
+		// If you want to use specific MediaPlayer, you can make your own object like GVRVideoObject.
+		
+		return videoObject;
+	}
+	
+	
+	
+	// 3. Create webview object
+	private GVRSceneObject buildWebViewObject(){
+
+		// Create your own WebView.
+		// This WebView follow Android WebView.
+		WebView webView = new WebView(mGVRContext.getContext());
```

Comment:
It must be exciting for the developers, as now getting those missing examples like GVRCameraSceneObject and GVRWebViewSceneObject. Thank you.

For buildWebViewObject(),
It must be changed to run all the WebView related methods to UI thread. NullPointerException or IllegalStateException otherwise. It has changed from Android 4.4 (API level 19).